### ISSUE TITLE
The "capture area" rectangle is not drawn

### ISSUE THREAD

**Issue by nicoulaj** [zachtib/lookit#1]
"Capture area" works fine but no rectangle is drown while doing the selection.

Ubuntu lucid / GNOME / Compiz activated.

**Comment by zachtib** [comment]
This appears to be a problem specific to when compiz is running, as I experience it intermittently with both lookit and gnome-screenshot when compiz is running.

Short of waiting on Compiz to update and magically fix the issue, the best bet is probably a different screengrabber for when a compositing window manager is active. Sadly, it's too late for this to make it in 0.3, but I'm going to try and have something ready for 0.4.

**Comment by vgarvardt** [comment]
Just updated to 0.4 - same problem.

**Comment by zachtib** [comment]
It's still an issue with composited desktops. Ill have to write a separate
rectangle drawer for compiz and the like, it just didn't make it in this
release.
On Apr 25, 2011 3:47 AM, "vgarvardt" <
reply@reply.github.com>
wrote:

**Comment by will-stone** [comment]
Same problem, Natty + Unity. Otherwise very good, thanks.

**Comment by craige111** [comment]
Yeah this bug is affecting me too on Natty / Unity and makes the selection screenshot feature largely unusable.

**Comment by zachtib** [comment]
Ok, I'm finally working on the new cairo selector, which will make it into 1.1 (for real this time). The selector itself was just checked in, but Lookit doesn't actually use it yet, as it will break if compositing is not enabled, so I need to account for that.

**Comment by zachtib** [comment]
Update: the latest commit enables the new selector, but the non-composited fallback isn't in place yet

**Comment by zachtib** [comment]
This should now be fixed in the master branch. I've added the new cairo based selector which works for both composited and non-composited desktops

**Comment by craige111** [comment]
This bug is still not fixed. I got the newest build from the PPA today and when selecting the rectangle, the screen goes dark, but when you drag a selection, it does not show it as normal again in the rectangle, and there is no rectangle drawn.  You get a dark screenshot eventually.

**Comment by zachtib** [comment]
What's your distro / window manager?

**Comment by zachtib** [comment]
Everyone, I want this to work, so if the new Cairo selector isn't working for you, please post the following information:

Linux distro and version (ie, Ubuntu 11.04)
Desktop environment (Gnome 2 / Unity / Gnome 3)
Window manager (Compiz/Metacity/Mutter)
Video card and Driver
Number of monitors

and anything else you can think of that could be related, and I'll look into it

**Comment by craige111** [comment]
Ubuntu 11.04
Unity
Compiz
Intel graphics  (driver unknown)
1 Monitor

**Comment by zachtib** [comment]
It looks like it's an issue with Intel graphics, a friend of mine has confirmed the issue on his computer. Looking into it.

**Comment by nicoulaj** [comment]
Ubuntu 11.04, Unity, Compiz, ATI card with open source driver: works fine.

**Comment by zachtib** [comment]
thanks, nicoulaj. Seems the issue is isolated to Intel graphics then

**Comment by craige111** [comment]
With the newest build the behaviour is different, and a bit of rectangle is drawn on Intel but there is a delay and does not start from where you start dragging.  The screenshot you get out is partially shaded.

**Comment by zachtib** [comment]
So you know: I'm pushing out 1.1 with the new selector. I'm still trying to figure out what causing this issue with Intel cards (not having one myself makes it hard to test) but I'm including an option in the preferences dialog to force Lookit to use the non-composited selector, so if you have an issue with it, check that option in the preferences dialog. Marking this closed, and opening a new issue for the Intel-specific bug.
